lib/src/pipeline.rs

```diff
@@ -0,0 +1,88 @@
+/*
+ * Panopticon - A libre disassembler
+ * Copyright (C) 2017 Panopticon authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//! Disassembly-Analysis loop.
+
+use futures::sync::mpsc;
+use futures::{stream,Stream,Sink,Future};
+use std::thread;
+use std::collections::{
+    HashMap,
+};
+use std::iter::FromIterator;
+use {
+    Architecture,
+    Program,
+    Function,
+    Region,
+    CallTarget,
+    Rvalue,
+    ssa_convertion,
+};
+use graph_algos::{
+    VertexListGraphTrait,
+    GraphTrait,
+};
+use std::fmt::Debug;
+
+/// Starts disassembling insructions in `region` and puts them into `program`. Returns a stream of
+/// of newly discovered functions.
+pub fn pipeline<A: Architecture + Debug + 'static >(program: Program, region: Region, config: A::Configuration) -> Box<Stream<Item=Function,Error=()> + Send>
+where A::Configuration: Debug {
+    let (tx,rx) = mpsc::channel::<Function>(10);
+    thread::spawn(move || {
```

Comment:
The `thread::spawn` doc says:
>The join handle will implicitly detach the child thread upon being dropped.
So, the resources should be freed when the closure returns. I'd like to notify the UI when the disassembly is done, but that's not super important IMHO.